### ISSUE TITLE
Formula script PDF template helpers

### ISSUE THREAD

**Issue by radek3911** [espocrm/espocrm#2955]
**Is your feature request related to a problem? Please describe.**
If user wants to use advanced logic in PDF template, most of the time custom template helper must be programmed.

**Describe the solution you'd like**
Ability to create template helpers using formula script.

**Describe alternatives you've considered**
You can program template helpers, but my thinking is that it could be beneficial to leverage formula script to create them, as most template helpers are simple (usually some data processing from related entities - which formula script can handle)

**Additional context**
I noticed in commit history that you are working on reworking PDF templates to use ACE editor as wysiwyg can be faulty with advaned templating, which is very nice. I think this could be a great addition to 8.2.0 by making Espo "less-needed for programming" and more "clickable and configurable by user" thanks to this feature.

Thanks for consideration

**Comment by yurikuzn** [comment]
It's possible to create a read-only field that will be filled with formula. And then used within PDF template. There is no new features to be considered for v8.2, unless very small improvements.

**Comment by yurikuzn** [comment]
For information, currently  here's 530 items in the backlog assigned on me, it's only issues related to development.

**Comment by radek3911** [comment]
> It's possible to create a read-only field that will be filled with formula. And then used within PDF template. There is no new features to be considered for v8.2, unless very small improvements.

Sure this can work, thanks for all your hard work on EspoCRM.

**Comment by Kharg** [comment]
@yurikuzn 
Why not using a service such as uservoice so that users can vote for features or propose new features in a better way? It also serves as a Roadmap.
Or even better, a espocrm portal with such features. It would be extremely useful

**Comment by yurikuzn** [comment]
We already have a good amount of incoming requests, we have forum, emails, GitHub, conversations with customers. Do we need having more means when we are are already overwhelmed? New features not always mean a better product, features with most votes not always represent what is better for the business. I stated multiple times that had we closed all feature request means, we would have work to do for years.
